src/shader/camera.rs

```diff
@@ -1,14 +1,34 @@
 use std::f32::consts::PI;
-
 use vek::Mat4;
-
+use vek::Vec4;
+use serde::{Serialize, Deserialize};
 use crate::config;
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(default)]
+#[serde(deny_unknown_fields)]
 pub struct Camera {
-    //TODO(#4): These fields should be replaced with: position, target, fov, aspect_ratio_x, etc.
-    pub(crate) view: Mat4<f32>,
-    pub(crate) projection: Mat4<f32>,
+    pub(crate) eye: Vec4<f32>,
+    pub(crate) target: Vec4<f32>,
+    pub(crate) fovy: f32,
+    pub(crate) aspect_ratio_x: f32,
+    pub(crate) aspect_ratio_y: f32,
+    pub(crate) near: f32,
+    pub(crate) far: f32,
+}
+
+impl Default for Camera {
+    fn default() -> Camera {
+        Camera {
+            eye: Vec4::new(0.0,1.0,8.5,1.0),
```

Comment:
```suggestion
            eye: Vec4 {x: 0.0, y: 1.0, z: 8.5, w: 1.0},
```

Since [all the fields of Vec4 are public](https://docs.rs/vek/0.9.8/vek/vec/repr_c/vec4/struct.Vec4.html), you can actually construct it using its fields directly. This has the advantage of not requiring the reader to remember which argument refers to which field. :)

(There should be a button you can use to instantly apply this and other suggested changes)